src/main.cpp

```diff
@@ -43,7 +43,7 @@ int main(int argc, char ** argv) {
 	return 0;
 }
 
-#ifdef WIN32
+#ifdef _WIN32
```

Comment:
use WIN32 because that is what it should be

Reply:
_WIN32 is the actual one that is defined by the msvc compiler: https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-160

Reply:
We use WIN32 as a generic way to specify that we're compiling for windows.
_WIN32 could probably be for parts which are necessary for MSVC to function or we could just rename WIN32 again.

Reply:
I haven't ran it on mingw, so can't say what would be the best way to abstract the differences between all the compilers. The most obvious problem was headers, as mingw is compliant with linux and as such when one breaks, the other follows.